picard/util/pipe.py

```diff
@@ -175,14 +175,14 @@ def read_from_pipe(self, timeout_secs: Optional[float] = None) -> List[str]:
         try:
             res = reader.result(timeout=timeout_secs)
             if res:
-                out = [r for r in res.split(self.MESSAGE_TO_IGNORE) if r]
+                out = set([r for r in res.split(self.MESSAGE_TO_IGNORE) if r])
```

Comment:
This looks very weird to me, returning a `set` means:

- identical messages are reduced to one
- order of messages isn't preserved

Though not sure I'm misunderstanding this code, can you explain?

BTW, instead of `set([comprehension])` one can write `{comprehension}`

Reply:
The two points you've made was the actual reason I've used such solution. But if you're saying that users will care about the order of files or they want duplicates, I may rollback to list, no problem.

Reply:
It isn't about files (which are a specific case, related to our usage), but the overall logic of a fifo pipe, one expects order to be preserved (as well as duplicates).